src/analyses/uninit.ml

```diff
@@ -91,56 +83,42 @@ struct
     List.fold_left f [] (access_one_byval a false rval)
 
   let vars a (rval:exp) : Addr.t list =
-    List.map Addr.from_var_offset (varoffs a rval)
-
-  let is_prefix_of (v1,ofs1: varinfo * (Addr.field,Addr.idx) Lval.offs) (v2,ofs2: varinfo * (Addr.field,Addr.idx) Lval.offs) : bool =
-    let rec is_offs_prefix_of pr os =
-      match (pr, os) with
-      | (`NoOffset, _) -> true
-      | (`Field (f1, o1), `Field (f2,o2)) -> CilType.Fieldinfo.equal f1 f2 && is_offs_prefix_of o1 o2
-      | (_, _) -> false
-    in
-    CilType.Varinfo.equal v1 v2 && is_offs_prefix_of ofs1 ofs2
+    List.map Addr.of_mval (varoffs a rval)
 
+  let is_prefix_of m1 m2 = Option.is_some (Addr.Mval.prefix m1 m2)
 
   (* Does it contain non-initialized variables? *)
   let is_expr_initd a (expr:exp) (st:D.t) : bool =
     let variables = vars a expr in
-    let raw_vars = List.filter_map Addr.to_var_offset variables in
+    let raw_vars = List.filter_map Addr.to_mval variables in
```

Comment:
`mvals`?

Reply:
`Addr.to_mval` still returns one optional mval.

Reply:
Yes, I think i meant the `raw_vars` name on the left... I think what we call a `raw_var` here is precisely an `mval`.

Reply:
I looked at this and simplified it even more: this contained pointless back-and-forth conversions which fuse to identity.